cpp/src/buffer/buffer.cpp

```diff
@@ -33,14 +33,21 @@ Buffer::Event::~Event() {
 
 [[nodiscard]] bool Buffer::Event::is_ready() {
     if (!done_.load(std::memory_order_relaxed)) {
-        bool result = cudaEventQuery(event_) == cudaSuccess;
+        bool result = (cudaEventQuery(event_) == cudaSuccess);
```

Comment:
Check for other errors like:
```suggestion
        ​cudaError_t err = cudaEventQuery(event_);
        if(err != cudaSuccess && err != cudaErrorNotReady) {
            RAPIDSMPF_CUDA_TRY(err);
        }
        bool result = (cudaEventQuery(event_) == cudaSuccess);
```

Reply:
Good catch, I previously assumed non-success means the event isn't done, but that's not always true.